embedded/tbtree/tbtree.go

```diff
@@ -1834,24 +1904,37 @@ func (n *innerNode) split() (node, error) {
 		nodes: n.nodes[splitIndex:],
 		mut:   true,
 	}
-	newNode.updateTs()
+
+	newNode.update()
 
 	n.nodes = n.nodes[:splitIndex]
 
-	n.updateTs()
+	n.update()
 
 	metricsBtreeInnerNodeEntries.WithLabelValues(n.t.path).Observe(float64(len(n.nodes)))
 	metricsBtreeInnerNodeEntries.WithLabelValues(n.t.path).Observe(float64(len(newNode.nodes)))
 
 	return newNode, nil
 }
 
-func (n *innerNode) updateTs() {
+func (n *innerNode) update() {
 	n._ts = 0
+	minOff := int64(math.MaxInt64)
+
 	for i := 0; i < len(n.nodes); i++ {
 		if n.ts() < n.nodes[i].ts() {
 			n._ts = n.nodes[i].ts()
 		}
+
+		if minOff > n.nodes[i].minOffset() && !n.nodes[i].mutated() {
+			minOff = n.nodes[i].minOffset()
+		}
+	}
+
+	if minOff == int64(math.MaxInt64) {
+		n._minOff = 0
```

Comment:
This is to cover the case when there are no child nodes?

Because otherwise, current node would have to be marked as mutated too and would be written by the next flush anyway.

Reply:
the node is already mutated.

if none of its children is serialized, then minOff will be set to 0